.github/CONTRIBUTOR_GUIDE/CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+
+Let's work together to keep our codebase clean, efficient, and maintainable! 💪🚀
+
+
+## Getting Started
+
+Before contributing, please ensure that you have the following setup:
+
+1. **Visual Studio Code** (recommended for script development)
+2. **Recommended VS Code Extensions:**
+   - [Shell Syntax](https://marketplace.visualstudio.com/items?itemName=bmalehorn.shell-syntax)
+   - [ShellCheck](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck)
+   - [Shell Format](https://marketplace.visualstudio.com/items?itemName=foxundermoon.shell-format)
+
+### Important Notes
+- Use [AppName.sh](https://github.com/community-scripts/ProxmoxVE/blob/contributor_guide/.github/CONTRIBUTOR_GUIDE/ct/AppName.sh) and [AppName-install.sh](https://github.com/community-scripts/ProxmoxVE/blob/contributor_guide/.github/CONTRIBUTOR_GUIDE/install/AppName-install.sh) as templates when creating new scripts.
+- The call to `community-scripts/ProxmoxVE` should be adjusted to reflect the correct fork URL.
+
+---
+
+# 🚀 The Application Script (ct/AppName.sh)
+
+- You can find all coding standards, as well as the structure for this file [here](https://github.com/community-scripts/ProxmoxVE/blob/contributor_guide/.github/CONTRIBUTOR_GUIDE/ct/AppName.md).
+- These scripts are responsible for container creation, setting the necessary variables and handling the update of the application once installed.
+
+---
+
+# 🛠 The Installation Script (install/AppName-install.sh)
+
+- You can find all coding standards, as well as the structure for this file [here](https://github.com/community-scripts/ProxmoxVE/blob/contributor_guide/.github/CONTRIBUTOR_GUIDE/install/AppName-install.md).
+- These scripts are responsible for the installation of the application.
+
+---
+
+## 🚀 Building Your Own Scripts
+
+Start with the [template script](https://github.com/community-scripts/ProxmoxVE/blob/contributor_guide/.github/CONTRIBUTOR_GUIDE/install/AppName-install.sh)
+
+---
+
+## 🤝 Contribution Process
+
+### 1. Fork the repository
+Fork to your GitHub account
+
+### 2. Clone your fork on your local environment 
+```bash
+git clone https://github.com/yourUserName/ForkName
+```
+
+### 3. Create a new branch
+```bash
+git switch -c your-feature-branch
+```
+
+### 4. Change paths in build.func install.func and AppName.sh
+To be able to develop from your own branch you need to change `https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main` to `https://raw.githubusercontent.com/[USER]/[REPOSITORY]/refs/head/[BRANCH]`. This change is only for testing. Before opening a Pull Request you should change this line change all this back to point to `https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main`.
```

Comment:
Overall this sounds like the typicall GitHub Flow style, which is reasonable.
Only Step 4 is special to this repository.

What do you think about this suggestion:
1. Making a small guide "Setting up your development environment" -> This is essentially Step 4
2. For the rest, refer to guides like https://github.blog/developer-skills/github/beginners-guide-to-github-creating-a-pull-request/

I think we will never do a better job than the Github docs in the regular content. I think we should keep the focus on the repo and create visibility on what is special here.

Reply:
We are thinking about the contribution procces for the Pull Requests. It is not really practical how it is done at the moment, as you can not test the functionality of the script besides running code from an unknown repo or coppying the code to your one repo for testing. I would leave this as is atm.